components/core/src/clp_s/search/OutputHandler.hpp

```diff
@@ -187,6 +197,50 @@ class ResultsCacheOutputHandler : public OutputHandler {
             QueryResultGreaterTimestampComparator>
             m_latest_results;
 };
+
+/**
+ * Output handler that performs an aggregation operation and can send the results to a reducer
+ */
+class ReducerOutputHandler : public OutputHandler {
+public:
+    // Constructor
+    ReducerOutputHandler(int socket_fd, bool should_output_timestamp, bool should_marshal_records)
```

Comment:
Why do we need `should_marshal_records`?

Reply:
So that local aggregation subclasses can specify whether they need records to be marshalled or not. Will remove this class anyway though per other comment.